test/integration/protocol_integration_test.cc

```diff
@@ -4417,6 +4417,7 @@ TEST_P(ProtocolIntegrationTest, LocalInterfaceNameForUpstreamConnection) {
 #endif
 
 TEST_P(DownstreamProtocolIntegrationTest, InvalidRequestHeaderName) {
+  config_helper_.addRuntimeOverride("envoy.reloadable_features.validate_upstream_headers", "false");
```

Comment:
sorry why doesn't disable client header validation disable validation?
given this runtime guard is slated for removal I think we need to actually fix things so we can not override it.
/wait

Reply:
I'm doing this in a separate PR. The boolean flag works for UHV but unfortunately did not fully disable validation in non UHV build. It needs more plumbing into the codec, which makes more sense for a separate PR.

Reply:
mind TODOing the follow-up?

Reply:
Done